x-pack/plugin/text-structure/src/test/java/org/elasticsearch/xpack/textstructure/structurefinder/GrokPatternCreatorTests.java

```diff
@@ -699,4 +701,19 @@ public void testValidateFullLineGrokPatternGivenInvalid() {
 
         assertEquals("Supplied Grok pattern [" + grokPattern + "] does not match sample messages", e.getMessage());
     }
+
+    public void testLongestRun() {
+
+        List<Integer> sequence = new ArrayList<>();
+        for (int before = randomIntBetween(0, 41); before > 0; --before) {
```

Comment:
It is kind of covered by the random, but you could add explicit test cases for:
- the longest sequence is the prefix
- the longest sequence is the suffix

Reply:
I made it more likely that these cases get tested by making each true 50% of the time.  I tested all the scenarios locally and I think one test with higher probability of testing the edge cases should be sufficient to prevent regressions.  There shouldn't be much need to change this code again.